File: output/gemma4-e4b/avatar/Python/Java/codeforces_629_A.java
import java.util.Scanner;

public class codeforces_629_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Read N
        if (!scanner.hasNextInt()) {
            // Handle case where N might not be provided
            return;
        }
        int n = scanner.nextInt();
        scanner.nextLine(); // Consume the rest of the line after reading N

        // Read the cake structure
        String[] cake = new String[n];
        for (int i = 0; i < n; i++) {
            if (scanner.hasNextLine()) {
                cake[i] = scanner.nextLine();
            } else {
                // Handle insufficient input lines
                return;
            }
        }
        scanner.close();

        double happiness = 0.0;

        // --- 1. Row Processing ---
        for (int i = 0; i < n; i++) {
            String line = cake[i];
            int chocCount = 0;
            for (int j = 0; j < line.length(); j++) {
                if (line.charAt(j) == 'C') {
                    chocCount++;
                }
            }

            if (chocCount > 1) {
                // Calculate nfat = chocCount * (chocCount - 1)
                // Using long for product calculation to prevent overflow
                long nfat = (long) chocCount * (chocCount - 1);
                happiness += (double) nfat / 2.0;
            }
        }

        // --- 2. Column Processing ---
        for (int j = 0; j < n; j++) { // j is the column index
            int chocCount = 0;
            for (int i = 0; i < n; i++) { // i is the row index
                // Check if the character at (i, j) is 'C'
                if (cake[i].length() > j && cake[i].charAt(j) == 'C') {
                    chocCount++;
                }
            }

            if (chocCount > 1) {
                // Calculate nfat = chocCount * (chocCount - 1)
                long nfat = (long) chocCount * (chocCount - 1);
                happiness += (double) nfat / 2.0;
            }
        }

        // Output the integer part of happiness
        System.out.println((int) happiness);
    }
}

File: output/gemma4-e4b/avatar/Java/Python/codeforces_629_A.py
import sys

def solve():
    # Read all input lines
    input_data = sys.stdin.read().split()
    
    if not input_data:
        return

    # The first element is num
    try:
        num = int(input_data[0])
    except ValueError:
        # Handle case where input might be malformed
        return

    # The remaining elements are the strings
    arr = input_data[1:num + 1]
    
    fine = 0
    
    # --- 1. Column-wise check ---
    # We assume all strings are at least length 'num' based on the original Java logic.
    # The number of columns to check is determined by the minimum length of the strings, 
    # but since the Java code iterates up to 'num', we stick to that limit.
    
    # Determine the actual number of columns to check (min length of all strings)
    # If we strictly follow the Java code, we iterate up to 'num'.
    num_cols = num
    
    for i in range(num_cols): # i is the column index
        counter = 0
        for k in range(num): # k is the string index
            # Check if the string arr[k] has a character at index i
            if i < len(arr[k]) and arr[k][i] == 'C':
                counter += 1
        
        # Calculate combinations: counter * (counter - 1) / 2
        fine += (counter * (counter - 1)) // 2
        
    # --- 2. Row-wise check ---
    for i in range(num): # i is the string index
        s = arr[i]
        counter = 0
        
        # Count 'C's in the current string s
        for k in range(len(s)): # k is the character index
            if s[k] == 'C':
                counter += 1
        
        # Calculate combinations
        fine += (counter * (counter - 1)) // 2
        
    print(fine)

solve()